src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -783,3 +783,118 @@ class JoinClauseSegment(BaseSegment):
+        # Nested Selects
+        Bracketed(Ref("SelectableGrammar")),
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class AliasExpressionSegment(BaseSegment):
+    """A reference to an object with an `AS` clause.
+
+    The optional AS keyword allows both implicit and explicit aliasing.
+    Note also that it's possible to specify just column aliases without aliasing the table as well:
+    .. code-block:: sql
+
+        SELECT * FROM VALUES (1,2) as t (a, b);
+        SELECT * FROM VALUES (1,2) as t;
+        SELECT * FROM VALUES (1,2) as (a, b);
+
+    """
+
+    type = "alias_expression"
+    match_grammar = Sequence(
+        Ref.keyword("AS", optional=True),
+        OneOf(
+            Sequence(
+                # maybe table alias
+                Ref("SingleIdentifierGrammar", optional=True),
+                # maybe column aliases
+                Bracketed(Ref("SingleIdentifierListSegment"), optional=True),
+            ),
+            Ref("QuotedLiteralSegment"),
+        ),
+    )
+
+
+@spark3_dialect.segment()
+class DelimitedValues(BaseSegment):
+    """A ``VALUES`` clause can be a sequence either of scalar values or tuple values.
+
+    We make no attempt to ensure that all records have the same number of columns besides the distinction between
+    all scalar or all tuple, so for instance ``VALUES (1,2), (3,4,5)`` will parse but is not legal SQL.
+
+
+    """
+
+    type = "delimited_values"
+    match_grammar = OneOf(Delimited(Ref("ScalarValue")), Delimited(Ref("TupleValue")))
+
+
+@spark3_dialect.segment()
+class ScalarValue(BaseSegment):
+    """An element of a ``VALUES`` clause that has a single column.
+
+    Ex: ``VALUES 1,2,3``
+    """
+
+    type = "scalar_value"
+    match_grammar = OneOf(
+        Ref("LiteralGrammar"),
+        Ref("BareFunctionSegment"),
+        Ref("FunctionSegment"),
+    )
+
+
+@spark3_dialect.segment()
+class TupleValue(BaseSegment):
+    """An element of a ``VALUES`` clause that has multiple columns.
+
+    Ex: ``VALUES (1,2), (3,4)``
+    """
+
+    type = "tuple_value"
+    match_grammar = Bracketed(
+        Delimited(
+            Ref("ScalarValue"),
+        )
+    )
+
+
+@spark3_dialect.segment(replace=True)
+class ValuesClauseSegment(BaseSegment):
+    """A ``VALUES`` clause, as typically used with ``INSERT`` or ``SELECT``.
+
+    The Spark SQL reference does not mention ``VALUES``` clauses except in the context of ``INSERT`` statements.
+    However, they appear to behave the same as in `postgres <https://www.postgresql.org/docs/13/sql-values.html>`.
+
+    In short, they can appear anywhere a ``SELECT`` can, and also as bare ``VALUES`` statements. Here are some examples:
+    .. code-block:: sql
+
+        VALUES 1,2 LIMIT 1;
+        SELECT * FROM VALUES (1,2) as t (a,b);
+        SELECT * FROM (VALUES (1,2) as t (a,b));
+        WITH a AS (VALUES 1,2) SELECT * FROM a;
+
+    """
+
+    type = "values_clause"
+    match_grammar = StartsWith(Ref.keyword("VALUES"))
```

Comment:
```suggestion
    match_grammar = StartsWith("VALUES")
```
`Ref.keyword` is implied here. The only time you need to use it is if you also need to pass an additional argument e.g. `Ref.keyword("VALUES", optional=True)`